main.go

```diff
@@ -108,32 +106,33 @@ func doCheckExistenceAndParse() {
 		}
 
 		purls, invalidPurls := packages.ExtractPurlsUsingDep(*project)
-		if len(invalidPurls) > 0 {
-			audit.LogInvalidSemVerWarning(config.NoColor, config.Quiet, invalidPurls)
-		}
 
-		var packageCount = len(purls)
-		checkOSSIndex(purls, packageCount)
+		checkOSSIndex(purls, invalidPurls)
 	case strings.Contains(config.Path, "go.sum"):
 		mod := packages.Mod{}
 		mod.GoSumPath = config.Path
 		if mod.CheckExistenceOfManifest() {
 			mod.ProjectList, _ = parse.GoSum(config.Path)
 			var purls = mod.ExtractPurlsFromManifest()
-			var packageCount = len(purls)
 
-			checkOSSIndex(purls, packageCount)
+			checkOSSIndex(purls, []string{})
```

Comment:
Pretty sure you can pass `nil` rather than instantiating `[]string`.

In the near future I'm going to likely move from `[]string` to `[]Purl`, as we can use this package to pass around our purls in a nicer way: https://github.com/package-url/packageurl-go, and we could add a `Valid bool` on to that as well, simplifying some of the code downstream.

Reply:
That sounds good....i still just put the nil slice in this commit b03d1aa